### PR TITLE
Fix clash between QDQ propagation and TransposeOptimizer

### PR DESCRIPTION
**Pull request by skottmckay**
**Description**: 
Prevent a loop between the QDQ propagation and Transpose optimizer by enabling phase 2 of the transpose optimizer and only running it once. This avoids the loop and produces an optimal model where a Transpose that is not part of a QDQ node group runs prior to a DQ node (so only has 8-bit instead of 32-bit data to transpose).

Disable the TransposeOptimizer in the QDQ propagation unit tests to simplify 

**Motivation and Context**
#11605 , #11537, #11686

### CODE REVIEW

**Review by yufenglee — APPROVED** [review]
:shipit:

File: onnxruntime/core/session/inference_session.cc [hunk 1]
```diff
@@ -924,7 +921,7 @@ common::Status InferenceSession::TransformGraph(onnxruntime::Graph& graph,
   // 4. insert copy nodes
   // 5. insert cast nodes.
 
-  // first apply global(execution provider independent),  level 1(default/system/basic) graph to graph optimizations
+  // first apply global(execution provider independent), level 1(default/system/basic) graph to graph optimizations
```
Comment (edgchen1) [inline onnxruntime/core/session/inference_session.cc:924]: I was confused by this comment. the next line applies Level1 transformers only (not TransformerLevel::Default ones).
how about something like this: `first apply execution provider independent, level 1 (basic) graph optimizations`

**Review by edgchen1 — COMMENTED** [review]
is it possible to have a test to verify the fix to the issue? e.g., testing that the model from the issue loads successfully

File: onnxruntime/core/session/inference_session.cc (hunk 1 above)
Comment (skottmckay) [reply onnxruntime/core/session/inference_session.cc:924]: I think it's confusing as there aren't actually any 'default' level transformers. In fact, 'default' is actually equivalent to 'none'

https://github.com/microsoft/onnxruntime/blob/006597b9b8d581c4db5e4294863a33c65cb49fb5/onnxruntime/core/session/abi_session_options.cc#L125-L127

I'll update the comment.

File: onnxruntime/core/session/inference_session.cc [hunk 2]
```diff
@@ -918,13 +915,13 @@ common::Status InferenceSession::TransformGraph(onnxruntime::Graph& graph,
                                                 SessionState& session_state,
                                                 bool saving_model_in_ort_format) {
   // The transformer order:
-  // 1. built-in graph rewriter
-  // 2. each execution provider's transformer
-  // 3. do node placement according to kernel definition
+  // 1. run level 1 optimizations. these only use ONNX operators.
+  // 2. partition nodes based on EP capabilities. EPs may fuse nodes during this process.
+  // 3. run all optimizations. level 2 and 3 optimizations use contrib ops.
```
Comment (edgchen1) [inline onnxruntime/core/session/inference_session.cc:920]: thanks for clarifying the steps. unrelated, but also noticed that steps 4 and 5 are done in the opposite order in the code (cast then copy)

**Review by edgchen1 — DISMISSED** [review]

**Comment by skottmckay** [comment]
> is it possible to have a test to verify the fix to the issue? e.g., testing that the model from the issue loads successfully

Added unit test that replicated the series of nodes in the problem model. Verified it looped without the changes, and did the right thing with the changes.

**Review by yufenglee — APPROVED** [review]
:shipit:

**Pull request merged**